CMakeLists.txt

```diff
@@ -50,26 +50,19 @@ if(DEFINED CMAKE_USE_CUDD)
             BUILD_COMMAND make
             )
 
-    if(EXISTS ${cudd_SOURCE_DIR}/Makefile)
-        add_custom_target(cudd-make
-                COMMAND make
-                WORKING_DIRECTORY ${cudd_SOURCE_DIR}
-                )
-    else()
-        add_custom_target(cudd-make
-                COMMAND ./configure && make
-                WORKING_DIRECTORY ${cudd_SOURCE_DIR}
-                )
+    if(NOT EXISTS ${cudd_SOURCE_DIR}/Makefile)
+        message(STATUS "Configuring Cudd-3.0.0")
+        execute_process(COMMAND ./configure WORKING_DIRECTORY ${cudd_SOURCE_DIR})
     endif()
-
-    add_library(cudd-cplusplus STATIC IMPORTED)
-    add_dependencies(cudd-cplusplus cudd-make)
-    set_property(TARGET cudd-cplusplus PROPERTY IMPORTED_LOCATION ${cudd_SOURCE_DIR}/cplusplus/.libs/libobj.a)
+    message(STATUS "Building Cudd-3.0.0")
+    execute_process(COMMAND make WORKING_DIRECTORY ${cudd_SOURCE_DIR})
```

Comment:
Hmm, how is it actually possible that CUDD is _not_ built at this time, given `download_project` above already has a `BUILD_COMMAND`? Or is the above `download_project` in fact invalid, if I read https://github.com/Crascit/DownloadProject/blob/master/DownloadProject.cmake correctly? That documentation says that `BUILD_COMMAND` and `CONFIGURE_COMMAND` are not permitted in `download_project`?!

Reply:
That explains why it wasn't working then, thanks! In that case I guess we could use ExternalProject directly https://cmake.org/cmake/help/v3.10/module/ExternalProject.html instead of DownloadProject

Reply:
I haven't managed to make ExternalProject work as I want (it seems `./configure` can't get executed) so I've just cleaned up the BUILD_COMMAND and CONFIGURE_COMMAND which have no effect.